activesupport/lib/active_support/message_verifiers.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require "active_support/messages/rotation_coordinator"
+
+module ActiveSupport
+  class MessageVerifiers < Messages::RotationCoordinator
+    ##
+    # :method: initialize
+    # :call-seq: initialize(&secret_generator)
+    #
+    # Initializes a new instance. +secret_generator+ must accept a salt, and
+    # return a suitable secret (string).
+
+    ##
+    # :method: []
+    # :call-seq: [](salt)
+    #
+    # Returns a MessageVerifier configured with a secret derived from the
+    # given +salt+, and options from #rotate. MessageVerifier instances will
+    # be memoized, so the same +salt+ will return the same instance.
+
+    ##
+    # :method: []=
+    # :call-seq: []=(salt, verifier)
+    #
+    # Overrides a MessageVerifier instance associated with a given +salt+.
+
+    ##
+    # :method: rotate
+    # :call-seq: rotate(**options)
+    #
+    # Adds +options+ to the list of viable options. Messages will be signed
+    # using the first added options. When verifying, however, each viable
```

Comment:
```suggestion
    # using the first added option. When verifying, however, each viable
```

Reply:
I believe "options" is correct here.  Each `options` Hash is added to a list of `options` Hashes.  The first `options` Hash in the list will be used to sign messages.  But when verifying messages, each `options` Hash in the list is tried.

When I wrote this method doc, I did struggle a bit in trying to be concise and consistent across different methods.  I didn't want the reader to have to juggle too many terms.

Does the following read any better?

> Adds `options` as an entry in the list of viable options. Messages will be signed using the first entry in the list. When verifying, however, each entry will be tried, in order, until one succeeds.

Reply:
Ah sorry @jonathanhefner I misunderstood what was happening here.

Your suggestion works, or perhaps simply: "Messages will be signed using the first _of the_ added options"

Reply:
@lewispb I took another pass at it.  Please let me know if it still needs improvement!